starboard/shared/libevent/socket_waiter_internal.cc

```diff
@@ -32,6 +32,13 @@
 
 namespace sbposix = starboard::shared::posix;
 
+#if (defined(_GNU_SOURCE) || defined(_POSIX_VERSION)) && \
+    !(PLAYSTATION_GENERATION <= 5)
```

Comment:
why do we gate to playstation_generation <= 5 instead of having it true for all versions? or is this something that's being deprecated in the future so any future versions won't require it?

Reply:
Playstation ( unfortunately ) sets _POSIX_VERSION, even though `pipe()` is required by the version of POSIX it claims to support. Otherwise i wouldn't add that extra condition at all.

I just set it so that it is true for our current ports, so in future whoever might work on the next gen will get a compile error here and they'll be forced to revisit it.